phaser/phase.py

```diff
@@ -319,11 +319,22 @@ def execute_batch_step(self, step):
 class PhaseRecords(UserList):
     def __init__(self, *args):
         super().__init__(*args)
+        # Slicing a UserList results in constructing a brand new list, which
```

Comment:
is UserList from somewhere else?

Reply:
`UserList` is a class in `collections` that is meant to be used when you want to subclass `dict`.